src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -136,12 +136,91 @@ export default class SourceFileLinter {
 			if (this.isUi5controlMetadataNode(node)) {
 				ts.forEachChild(node, visitMetadataNodes);
 			}
+		} else if ((ts.isPropertyAssignment(node) || ts.isPropertyDeclaration(node) ||
+			ts.isVariableDeclaration(node) || ts.isShorthandPropertyAssignment(node)) &&
+			["renderer", "render"].includes(node.name?.getText())) {
+			// TODO: Analyze only metadata properties
+			this.analyzeControlRenderer(node);
 		}
 
 		// Traverse the whole AST from top to bottom
 		ts.forEachChild(node, this.#boundVisitNode);
 	}
 
+	analyzeControlRenderer(node: ts.PropertyAssignment |
+		ts.PropertyDeclaration | ts.VariableDeclaration | ts.ShorthandPropertyAssignment,
+		rendererName = "renderer") {
+		// Analyze renderer property when it's an ObjectLiterExpression
+		// i.e. { renderer: {apiVersion: "2", render: () => {}} }
+		if ((ts.isPropertyAssignment(node) || ts.isPropertyDeclaration(node) || ts.isVariableDeclaration(node)) &&
+			node.initializer && ts.isObjectLiteralExpression(node.initializer)) {
+			const apiVersionNode = node.initializer?.properties.find((prop) => {
+				return ts.isPropertyAssignment(prop) &&
+					ts.isIdentifier(prop.name) &&
+					prop.name.text === "apiVersion";
+			});
+
+			let nodeToHighlight: ts.PropertyAssignment | ts.PropertyDeclaration |
+				ts.VariableDeclaration | undefined = undefined;
+			if (!apiVersionNode) { // No 'apiVersion' property
+				nodeToHighlight = node;
+			} else if (ts.isPropertyAssignment(apiVersionNode) &&
+				apiVersionNode.initializer.getText() !== "2") { // String value would be "\"2\""
+				nodeToHighlight = apiVersionNode;
+			}
+
+			if (nodeToHighlight) {
+				this.#reporter.addMessage(MESSAGE.NO_DEPRECATED_RENDERER, nodeToHighlight);
+			}
+		// Analyze renderer property when it's a function i.e. { renderer: () => {} }
+		} else if (
+			(ts.isPropertyAssignment(node) || ts.isPropertyDeclaration(node) || ts.isVariableDeclaration(node)) &&
+			node.initializer &&
+			(ts.isMethodDeclaration(node.initializer) || ts.isArrowFunction(node.initializer) ||
+				ts.isFunctionExpression(node.initializer)) &&
+				node.name?.getText() === rendererName) {
+			this.#reporter.addMessage(MESSAGE.NO_DEPRECATED_RENDERER, node);
+		} else if (
+			((ts.isPropertyAssignment(node) || ts.isPropertyDeclaration(node)) &&
+				node.initializer && ts.isIdentifier(node.initializer)) ||
+				ts.isShorthandPropertyAssignment(node)) {
+			let renderInitializer = ts.isShorthandPropertyAssignment(node) ?
+				node.name.getText() :
+				node.initializer!.getText();
+
+			let useSource = false;
+			const sourceFile = node.getSourceFile();
+
+			const findRenderDeclaration = (potentialDeclarations: ts.Node) => {
```

Comment:
What about using the typeChecker (`this.#checker.getTypeAtLocation`) instead to resolve the type?

Reply:
Also see: https://github.com/SAP/ui5-linter/pull/374/files

Reply:
Thanks!